subprojects/wrapper/src/main/java/org/gradle/wrapper/GradleWrapperMain.java

```diff
@@ -88,7 +89,7 @@ private static File wrapperJar() {
         if (!location.getScheme().equals("file")) {
             throw new RuntimeException(String.format("Cannot determine classpath for wrapper Jar from codebase '%s'.", location));
         }
-        return new File(location.getPath());
+        return Paths.get(location).toFile();
```

Comment:
We need to keep this working for older VMs that Gradle can't run on (e.g., Java 6).  Could you add some fallback behavior here?

e.g.,
```
        try {
            return Paths.get(location).toFile();
        } catch (NoClassDefFoundError e) {
            return new File(location.getPath());
        }
```

Reply:
Wouldn't the initialization of the class fail if it can't find the dependent type? Not 100% familiar with with Java's class loading.

Also, the wrapper still needs 6 compatibility? I get that in core, but the wrapper gets forked  basically immediately, right?

Reply:
It's the very first thing that gets invoked when using `gradlew` and where we have the least information about what the build might need.  We want to fail with a nice message from the Gradle runtime (e.g, "You need Java 8") vs blowing up in the script with unknown class version problems or, in this case, newer JDK APIs.